src/org/exist/collections/MutableCollection.java

```diff
@@ -1258,6 +1258,9 @@ private void storeXMLInternal(final Txn transaction, final DBBroker broker, fina
         if(info.isCreating()) {
             info.getTriggers().afterCreateDocument(broker, transaction, document);
         } else {
+            final StreamListener listener = broker.getIndexController().getStreamListener(document, StreamListener.ReindexMode.REPLACE_DOCUMENT);
```

Comment:
it going to be new `StreamListener`, not same as at `startReplaceDocument` ... lock wrong to me